rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/TuplespaceAlg.scala

```diff
@@ -43,18 +43,19 @@ object TuplespaceAlg {
           case Right(Some((continuation, dataList))) =>
             val rspaceMatchCost =
               dataList
-                .map(_.cost.map(CostAccount.fromProto(_)).getOrElse(CostAccount.zero))
+                .map(_.cost.map(CostAccount.fromProto(_)).getOrElse(CostAccount(0)))
                 .toList
+                .map(ca => ca.copy(cost = Cost(Integer.MAX_VALUE) - ca.cost))
```

Comment:
The 'fishy' comment should go here ^

> This looks veeeeery fishy and suggests to me we should actually be adding cost and have an explicit limit to compare against. What does it even mean to subtract cost from an arbitrary value? thinking

Reply:
An arbitrary value is the phlos that a user is willing to pay when deploying the code.